### DISCUSSION TITLE
How to separate out the CRD API

### DISCUSSION THREAD

**Discussion by Nebulazhang** [kubernetes-sigs/kubebuilder#2353] [Support] [answered]
How to separate out the api(CRD ) and controller code instead of in one project.
Only manually？
Other projects will use our CR and will create it by client-go.
If these projects import our code included controller code, they may meet mod version conflict and import so many unnecessary and complex dependencies

**Comment by camilamacedo86** [comment, marked as answer]
Hi @Nebulazhang,

Sorry, but it seems hard to follow up exactly what you are looking for:
- See that ideally each project should have only the CRDs/API that is owned created by it
- Also, you can use kind created by other projects (we call it external types) but you do not need to import the whole project

You might want to look in the content of the open PR: https://github.com/kubernetes-sigs/kubebuilder/pull/3055 (see the comments as well). It seems address your questions 

I am marking it as answered and closing 
However, feel free to re-open if you need to.
